assignments/week_4/week_4_exercises/exercise_4_1/exercise_4_1.md

```diff
@@ -0,0 +1,127 @@
+# Exercise 4.1
+This is the first exercise of week 4, and the first one involving ROS with C++!
+
+# Exercise Objective
+For this exercise you will be adding a subscriber to a ROS node. The end result will be a node which
+subscribes to `/string_topic`, and prints the messages data to the ROS console.
+
+Adding a subscriber involves:
+ 
+ * Including the correct message header
+ * Making a callback function
+ * Creating the subscriber object
+ 
+To test that your subscriber works, you will:
+ * Run the node using `roslaunch`
+ * Find the log file created when the node was running
+ * Turn in the node and log file to show your node worked
+ 
+# Message Headers
+To use message types, you need to include the correct header.
+In this exercise, we will use the `String` message type from the `std_msgs` package.
+To include a message, add an include message at the top of the file, like this:
```

Comment:
I would add a sentence to explain that all ROS includes are in this format... or at least most?